software/base/src/main/java/brooklyn/entity/basic/SoftwareProcessImpl.java

```diff
@@ -429,8 +432,21 @@ protected final void callStartHooks() {}
             if (PortRange.class.isAssignableFrom(k.getType())) {
                 PortRange p = (PortRange)getConfig(k);
                 if (p != null && !p.isEmpty()) ports.add(p.iterator().next());
+            }   
+        }
+        
+        Map<String, Object> provisioningProperties = getConfig(PROVISIONING_PROPERTIES);
+        if(provisioningProperties.containsKey("required.ports")){
```

Comment:
The other keys in `PROVISIONING_PROPERTIES` correspond to `ConfigKey`s, such as those in `CloudLocationConfig`. It would be good to include `required.ports` as a `ConfigKey` there, rather than as a hard-coded string. If we do that, then it also needs to be respected by `JcloudsLocation` etc.

Also, there is a `CloudLocationConfig.INBOUND_PORTS` config key. If you set that in `PROVISIONING_PROPERTIES` then it (unfortunately but understandably) overrides all other values from the `getRequiredOpenPorts`. We could augment this with a `ADDITIONAL_INBOUND_PORTS` config key perhaps?